src/ostree/ot-admin-kargs-builtin-edit-in-place.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2014 Colin Walters <walters@verbum.org>
+ * Copyright (C) 2022 Huijing Hei <hhei@redhat.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; either version 2 of the licence or (at
+ * your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General
+ * Public License along with this library. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+
+#include "ot-main.h"
+#include "ot-admin-kargs-builtins.h"
+
+#include "ostree.h"
+#include "otutil.h"
+
+static char **opt_kargs_edit_in_place_append;
+
+static GOptionEntry options[] = {
+  { "append-if-missing", 0, 0, G_OPTION_ARG_STRING_ARRAY, &opt_kargs_edit_in_place_append, "Append kernel arguments if they do not exist", "NAME=VALUE" },
+  { NULL }
+};
+
+gboolean
+ot_admin_kargs_builtin_edit_in_place (int argc, char **argv, OstreeCommandInvocation *invocation, GCancellable *cancellable, GError **error)
+{
+  gboolean ret = FALSE;
+  guint i;
+  OstreeDeployment *deployment = NULL;
+  g_autoptr(GPtrArray) deployments = NULL;
+  g_autoptr(GOptionContext) context = NULL;
+  g_autoptr(OstreeSysroot) sysroot = NULL;
+  g_autoptr(OstreeKernelArgs) kargs = NULL;
+
+  context = g_option_context_new ("ARGS");
+
+  if (!ostree_admin_option_context_parse (context, options, &argc, &argv,
+                                          OSTREE_ADMIN_BUILTIN_FLAG_SUPERUSER,
+                                          invocation, &sysroot, cancellable, error))
+    goto out;
+
+  deployments = ostree_sysroot_get_deployments (sysroot);
+  if (deployments->len == 0)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "Unable to find a deployment in sysroot");
+      goto out;
+    }
```

Comment:
I think this is fine, though...I wonder if the "no deployments" case should actually be an error or just a silent no-op.  Mmm...I guess it makes sense to error out.  But it also seems valid to no-op.

Reply:
Not quite sure about this, maybe this can exit early before going into loop if no deployments?

Reply:
Yeah, could just `return TRUE;`.

That said...I honestly don't have a strong opinion on this.  It's fine by me to keep it as an error too.

Reply:
Update code style according to https://github.com/ostreedev/ostree/pull/2649